src/main/java/com/ldtteam/structurize/commands/PasteCommand.java

```diff
@@ -0,0 +1,225 @@
+        Manager.addToQueue(new TickedWorldOperation(instantPlacer, player));
+
+        source.sendFailure(Component.translatable(PASTE_SUCCESS_MESSAGE));
+        return 1;
+    }
+
+    private static int onExecute(final CommandContext<CommandSourceStack> context) throws CommandSyntaxException
+    {
+        final BlockPos pos = BlockPosArgument.getSpawnablePos(context, POS);
+        final String packName  = StringArgumentType.getString(context, PACK_NAME);
+        final String path  = StringArgumentType.getString(context, FILE_PATH);
+
+        return execute(context.getSource(), pos, packName, path, Rotation.NONE, false, true);
+    }
+
+    private static int onExecuteWithRotation(final CommandContext<CommandSourceStack> context) throws CommandSyntaxException
+    {
+        final BlockPos pos = BlockPosArgument.getSpawnablePos(context, POS);
+        final String packName  = StringArgumentType.getString(context, PACK_NAME);
+        final String path  = StringArgumentType.getString(context, FILE_PATH);
+        final Rotation rotation = Rotation.values()[IntegerArgumentType.getInteger(context, ROTATION)];
+
+        return execute(context.getSource(), pos, packName, path, rotation, false, true);
+    }
+
+    private static int onExecuteWithRotationAndMirror(final CommandContext<CommandSourceStack> context) throws CommandSyntaxException
+    {
+        final BlockPos pos = BlockPosArgument.getSpawnablePos(context, POS);
+        final String packName  = StringArgumentType.getString(context, PACK_NAME);
+        final String path  = StringArgumentType.getString(context, FILE_PATH);
+        final Rotation rotation = Rotation.values()[IntegerArgumentType.getInteger(context, ROTATION)];
+        final boolean mirror = BoolArgumentType.getBool(context, MIRROR);
+
+        return execute(context.getSource(), pos, packName, path, rotation, mirror, true);
+    }
+
+    private static int onExecuteWithFull(final CommandContext<CommandSourceStack> context) throws CommandSyntaxException
+    {
+        final BlockPos pos = BlockPosArgument.getSpawnablePos(context, POS);
+        final String packName  = StringArgumentType.getString(context, PACK_NAME);
+        final String path  = StringArgumentType.getString(context, FILE_PATH);
+        final Rotation rotation = Rotation.values()[IntegerArgumentType.getInteger(context, ROTATION)];
+        final boolean mirror = BoolArgumentType.getBool(context, MIRROR);
+        final boolean pretty = BoolArgumentType.getBool(context, PRETTY);
+
+        return execute(context.getSource(), pos, packName, path, rotation, mirror, pretty);
+    }
+
+    protected static LiteralArgumentBuilder<CommandSourceStack> build()
+    {
+        return newLiteral(commandName)
+          .then(newArgument(POS, BlockPosArgument.blockPos())
+            .then(newArgument(PACK_NAME, StringArgumentType.word())
+              .then(newArgument(FILE_PATH, StringArgumentType.word())
```

Comment:
These should use the `string` argument type, so you can write a quoted string with spaces instead of doing the dot-replacement thing.  That's just weird.

Reply:
didn't know that was a thing